symphony/bdk/core/activity/form.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from symphony.bdk.core.activity.api import AbstractActivity, ActivityContext
+from symphony.bdk.gen.agent_model.v4_initiator import V4Initiator
+from symphony.bdk.gen.agent_model.v4_symphony_elements_action import V4SymphonyElementsAction
+
+logger = logging.getLogger(__name__)
+
+
+class FormReplyContext(ActivityContext[V4SymphonyElementsAction]):
+    """
+    Default implementation of the :py:class`ActivityContext` handled by the :py:class:`FormReplyActivity`
+    """
+
+    def __init__(self, initiator: V4Initiator, source_event: V4SymphonyElementsAction):
+        self._form_id = ""
+        self._form_values = {}
+        super().__init__(initiator, source_event)
+
+    @property
+    def form_id(self) -> str:
+        return self._form_id
+
+    @property
+    def form_values(self) -> dict:
+        return self._form_values
+
+    def get_form_value(self, field_name: str) -> str:
+        """Get the value of specified form field
+
+        :param field_name: the form field name
+        :return: the form field value if it is present, None otherwise
+        """
+        return self.form_values.get(field_name)
+
+
+class FormReplyActivity(AbstractActivity[FormReplyContext]):
+    """
+    A form reply activity corresponds to an Elements form submission.
+    """
+
+    def matches(self, context: FormReplyContext) -> bool:
+        pass
+
+    def on_activity(self, context: FormReplyContext):
+        pass
+
+    def before_matcher(self, context: FormReplyContext):
+        context._form_id = context.source_event.form_id
```

Comment:
I'd prefer `context._form_id` and `context._form_values` to be set in the constructor of `FormReplyContext` directly

Reply:
as the event is coming from the outside should we be defensive about what we get? (like if it is null or does not have a form id)

Reply:
> I'd prefer `context._form_id` and `context._form_values` to be set in the constructor of `FormReplyContext` directly

true, I wonder why we have the before_matcher() if we can do part of the context's constructor

Reply:
I tried sending a form without an `id` and It didn't work: ` {"code":400,"message":"Failed to parse MessageML: The attribute \"id\" is required"} so is that possible ?

I thought it was to separate the logic of modifying the context in `before_matcher()` before validating it in `matches()`

To handle the possible unwanted values we get, maybe we can do this, WDT @symphony-elias :
```Python
self._form_id = getattr(source_event, "form_id", None)
or
self._form_id = getattr(source_event, "form_id", "")
```
We definitely can add those to the constructor. Maybe I am missing the purpose of thibault setting them in `before_matcher()`